app.py

```diff
@@ -43,23 +43,69 @@
 
 @cache.memoize()
 def get_config_data():
-    """Fetches and returns data from the configuration options file as a dictionary."""
-    with open("config.json", "r") as f:
-        config = json.load(f)
+    """Loads user-defined configuration options.
+
+    The function first sets default configurations.
+    Then, it tries to load configurations from a .json file.
+    Then, it tries to load configurations from environment variables.
+
+    Returns:
+    dict: A dictionary containing user-defined configuration options.
+    """
+
+    # Set default configuration
+    config = {
+        "user": None,
+        "password": None,
+        "host": "127.0.0.1",
+        "port": 3306,
+        "database": None,
+        "url_base_pathname": "/dashboard/",
+        "query_row_limit": 5000,
+        "table_names_map": {
+            "game_info": None,
+            "game_content_info": None,
+            "game_player_info": None
+        }         
+    }
+
+    # Try to load configuration from .json file
+    if os.path.isfile("config.json"):
+        with open("config.json", "r") as file:
+            file_config = json.load(file)
+            # Only update config with values that are not None and exist in the default configuration
+            config.update(
+                {
+                    key: value
+                    for key, value in file_config.items()
+                    if value is not None and key in config
+                }
+            )
+
+    # Try to load configuration from environment variables
+    env_config = {
+        "user": os.environ.get("DB_USER"),
+        "password": os.environ.get("DB_PASSWORD"),
+        "host": os.environ.get("DB_HOST"),
+        "port": int(os.environ.get("DB_PORT"))
+        if os.environ.get("DB_PORT")
+        else None,
```

Comment:
This wasn't introduced here but to me it just looks odd.
```suggestion
        "port": int(os.environ.get("DB_PORT")) if os.environ.get("DB_PORT") else None,
```

Reply:
Just committed https://github.com/wesnoth/wesnoth-multiplayer-data-dashboard/pull/17/commits/b83005e83352328bb05cca8390852bd88e146a2f, a code formatting pass which resolves this thread.